### ISSUE TITLE
Module subpathing required to import types

### ISSUE THREAD

**Issue by yo1dog** [pubnub/javascript#411]
To import types, one must import them from module subpaths. This ties projects to the module's internal file structure, which is against best practices, and could make any internal file renaming/restructuring a breaking change.

Instead, you could expose the types from the module root using a namespace as described in https://github.com/pubnub/javascript/issues/410.

For example, currently one must:
```typescript
import type {FetchedMessage} from 'pubnub/lib/types/core/types/api/history';
```

Using the namespace you could instead do:
```typescript
declare namespace PubNub {
  export * from '../core/types/api/history';
  ...
}
```
```typescript
import {FetchedMessage} from 'pubnub';
```

Or, keeping the types namespaced:
```typescript
declare namespace PubNub {
  export * as History from '../core/types/api/history';
  ...
}
```
```typescript
import * as PubNub from 'pubnub';
let myMessages: PubNub.History.FetchedMessage[];
```

**Comment by parfeon** [comment]
Created a script which will merge all type definitions created by `tsc` into a single file. 
Common types (for whole library) part of the root `PubNub` namespace and specific API grouped under respective nested namespaces.

```ts
import PubNub, {PubNubConfiguration, Publish, History} from 'pubnub';

// PubNub client configuration.
let configuration: PubNubConfiguration = {
  publishKey: 'secret', 
  subscribeKey: 'secret', 
  userId: 'user-1'
}; 

// Fetch messages configuration.
let fetchParameter: History.FetchMessagesParameters = {
  channels: ['channel-1']
};
```

**Comment by parfeon** [comment]
This issue has been addressed in [8.2.10](https://github.com/pubnub/javascript/releases/tag/v8.2.10).
